src/core/syscall_info.rs

```diff
@@ -0,0 +1,138 @@
+use crate::core::errors::syscall_hadler_errors::SyscallHandlerError;
+use crate::core::errors::syscall_hadler_errors::SyscallHandlerError::*;
+use cairo_rs::serde::deserialize_program::*;
+use num_bigint::BigInt;
+use std::collections::HashMap;
+use std::fs;
+use std::path::Path;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub(crate) enum SyscallType {
+    EmitEvent {
+        selector: Option<Member>,
+        keys_len: Option<Member>,
+        keys: Option<Member>,
+        data_len: Option<Member>,
+        data: Option<Member>,
```

Comment:
Why are all these `Option<Member>` if they seem mandatory? Can't we fail at an early stage if whatever is supposed to provide them doesn't?